### PR TITLE
doc/start: update "memory" in hardware-recs.rst

### PR DESCRIPTION
**Pull request by zdover23**
This PR corrects some usage errors in the "Memory" section
of the hardware-recommendations.rst file. It also closes
some opened but never closed parentheses.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/start/hardware-recommendations.rst [hunk 1]
```diff
@@ -75,30 +75,30 @@ Memory
 ======
 
 Bluestore uses its own memory to cache data rather than relying on the
-operating system page cache.  In bluestore you can adjust the amount of memory
-the OSD attempts to consume with the :confval:`osd_memory_target` configuration
-option.
+operating system's page cache. In bluestore you can adjust the amount of memory
```
Comment (anthonyeleven) [inline doc/start/hardware-recommendations.rst:78]: Bluestore

File: doc/start/hardware-recommendations.rst [hunk 2]
```diff
@@ -75,30 +75,30 @@ Memory
...
-  extremely slow performance.
+  recommended (Ceph may fail to keep the memory consumption under 2GB and 
+  this may cause extremely slow performance).
 
 - Setting the memory target between 2GB and 4GB typically works but may result
   in degraded performance: metadata may be read from disk during IO unless the
   active data set is relatively small.
 
-- 4GB is the current default :confval:`osd_memory_target` size.  This default
+- 4GB is the current default :confval:`osd_memory_target` size. This default
```
Comment (anthonyeleven) [inline doc/start/hardware-recommendations.rst:90]: :p

File: doc/start/hardware-recommendations.rst [hunk 3]
```diff
@@ -75,30 +75,30 @@ Memory
...
   was chosen for typical use cases, and is intended to balance memory
   requirements and OSD performance for typical use cases.
```
Comment (anthonyeleven) [inline doc/start/hardware-recommendations.rst:92]: `typical use cases` is in this sentence twice.  Suggest truncating after `performance` while you're in this file.

File: doc/start/hardware-recommendations.rst [hunk 4]
```diff
@@ -110,8 +110,9 @@ option.
    needed, depending on the exact configuration of the system.
 
 When using the legacy FileStore backend, the page cache is used for caching
-data, so no tuning is normally needed, and the OSD memory consumption is
-generally related to the number of PGs per daemon in the system.
+data, so no tuning is normally needed. When using the legacy FileStore backend,
+the OSD memory consumption is related to the number of PGs per daemon in the
+system.
```
Comment (anthonyeleven) [inline doc/start/hardware-recommendations.rst:115]: Suggest `When using the legacy Filestore back end, OSD memory consumption is related to the number of PGs per daemon.`

When my book was in the editing phase, there was some controversy re `backend` and `back end`; the conclusion was to prefer the latter.

**Review by anthonyeleven — CHANGES_REQUESTED** [review]

File: doc/start/hardware-recommendations.rst (hunk 1 above)
Comment (zdover23) [reply doc/start/hardware-recommendations.rst:78]: Alright.

File: doc/start/hardware-recommendations.rst (hunk 4 above)
Comment (zdover23) [reply doc/start/hardware-recommendations.rst:115]: Alright.

File: doc/start/hardware-recommendations.rst (hunk 3 above)
Comment (zdover23) [reply doc/start/hardware-recommendations.rst:92]: Good point.

File: doc/start/hardware-recommendations.rst (hunk 1 above)
Comment (anthonyeleven) [reply doc/start/hardware-recommendations.rst]: Wait, I meant BlueStore.

**Review by anthonyeleven — APPROVED** [review]

**Comment by zdover23** [comment]
tick tock, API tests.

**Review by markhpc — COMMENTED** [review]
LGTM

**Pull request merged**